nengo/spa/tests/test_bind.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+
+import nengo
+from nengo import spa
+
+
+def test_basic():
+    with spa.SPA() as model:
+        model.bind = spa.Bind(dimensions=16)
+
+    inputA = model.get_module_input('bind_A')
+    inputB = model.get_module_input('bind_B')
+    output = model.get_module_output('bind')
+    # all nodes should be acquired correctly
+    assert inputA[0] is model.bind.A
+    assert inputB[0] is model.bind.B
+    assert output[0] is model.bind.output
+    # all inputs and outputs should share the same vocab
+    assert inputA[1] is inputB[1]
+    assert inputA[1].dimensions == 16
+    assert output[1].dimensions == 16
+
+
+def test_run(Simulator, seed):
+    vocab = spa.Vocabulary(16)
+    with spa.SPA(seed=seed) as model:
+        model.bind = spa.Bind(dimensions=32, vocab=vocab)
```

Comment:
Ah, that was a silly commit. Sorry about that. I got frustrated and didn't
test locally before seeing how it would run on TravisCI. I'll try the seed
thing that you recommended and get back to you.

On Wed, Sep 23, 2015 at 11:06 PM, Jan Gosmann <notifications@github.com>
wrote:

> In nengo/spa/tests/test_bind.py
> <https://github.com/nengo/nengo/pull/849#discussion_r40281932>:
>
> > +    inputB = model.get_module_input('bind_B')
> > +    output = model.get_module_output('bind')
> > +    # all nodes should be acquired correctly
> > +    assert inputA[0] is model.bind.A
> > +    assert inputB[0] is model.bind.B
> > +    assert output[0] is model.bind.output
> > +    # all inputs and outputs should share the same vocab
> > +    assert inputA[1] is inputB[1]
> > +    assert inputA[1].dimensions == 16
> > +    assert output[1].dimensions == 16
> > +
> > +
> > +def test_run(Simulator, seed):
> > +    vocab = spa.Vocabulary(16)
> > +    with spa.SPA(seed=seed) as model:
> > +        model.bind = spa.Bind(dimensions=32, vocab=vocab)
>
> How can you assign a 16-dimensional vocab to something that requires 32
> dimensions? Shouldn't this raise an error?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo/pull/849/files#r40281932>.
>